homeassistant/components/smtp/config_flow.py

```diff
@@ -0,0 +1,197 @@
+    CONF_ENCRYPTION,
+    CONF_SENDER_NAME,
+    CONF_SERVER,
+    DEFAULT_DEBUG,
+    DEFAULT_ENCRYPTION,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    ENCRYPTION_OPTIONS,
+)
+from .notify import MailNotificationService
+
+_LOGGER = logging.getLogger(__name__)
+
+
+BASE_DATA_FIELDS = {
+    vol.Optional(CONF_NAME): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.TEXT)
+    ),
+    vol.Required(CONF_RECIPIENT): SelectSelector(
+        SelectSelectorConfig(options=[], multiple=True, custom_value=True)
+    ),
+    vol.Required(CONF_SENDER): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.EMAIL)
+    ),
+    vol.Required(CONF_SERVER, default=DEFAULT_HOST): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.TEXT)
+    ),
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): NumberSelector(
+        NumberSelectorConfig(min=1, max=65535, mode=NumberSelectorMode.BOX, step=1)
+    ),
+    vol.Optional(CONF_ENCRYPTION, default=DEFAULT_ENCRYPTION): SelectSelector(
+        SelectSelectorConfig(options=ENCRYPTION_OPTIONS, translation_key="encryption")
+    ),
+    vol.Optional(CONF_USERNAME): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.TEXT)
+    ),
+    vol.Optional(CONF_PASSWORD): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.PASSWORD)
+    ),
+    vol.Optional(CONF_SENDER_NAME): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.TEXT)
+    ),
+}
+
+ADVANCED_DATA_FIELDS = {
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): NumberSelector(
+        NumberSelectorConfig(min=1, max=60, mode=NumberSelectorMode.BOX)
+    ),
+    vol.Optional(CONF_DEBUG, default=DEFAULT_DEBUG): BooleanSelector(
+        BooleanSelectorConfig()
+    ),
+    vol.Optional(CONF_VERIFY_SSL, default=True): BooleanSelector(
+        BooleanSelectorConfig()
+    ),
+}
+
+CONFIG_SCHEMA = vol.Schema(BASE_DATA_FIELDS | ADVANCED_DATA_FIELDS)
+
+UNIQUE_ENTRY_KEYS = [CONF_RECIPIENT, CONF_SENDER, CONF_SERVER, CONF_NAME]
+
+
+def validate_smtp_settings(settings: dict[str, Any], errors: dict[str, str]) -> None:
+    """Validate SMTP connection settings."""
+    try:
+        for recipient in cv.ensure_list(settings[CONF_RECIPIENT]):
+            vol.Email()(recipient)
+    except vol.Invalid:
+        errors[CONF_RECIPIENT] = "invalid_email_address"
+    try:
+        vol.Email()(settings[CONF_SENDER])
+    except vol.Invalid:
+        errors[CONF_SENDER] = "invalid_email_address"
+    if settings.get(CONF_USERNAME) and not settings.get(CONF_PASSWORD):
+        errors[CONF_PASSWORD] = "username_and_password"
+    if settings.get(CONF_PASSWORD) and not settings.get(CONF_USERNAME):
+        errors[CONF_USERNAME] = "username_and_password"
+
+    if errors:
+        return
+    settings[CONF_PORT] = cv.positive_int(settings[CONF_PORT])
+    settings[CONF_TIMEOUT] = cv.positive_int(settings[CONF_TIMEOUT])
+
+    service_class = MailNotificationService(
```

Comment:
We shouldn't use platform details outside the platform. This needs to be refactored.